### PR TITLE
[infra] task: updated dockerfile, fixed setup_java, minor fixes

### PR DESCRIPTION
**Pull request by yilmazbahadir**
Problem: 
- Dockerfile needed for containerization 
- Lack of cross environment support in `setup_java9.sh`

Solution: 
- Updated Dockerfile for multistage build, added user w/ UID and UNAME args and fixed the issues
- ~~Updated `setup_java9.sh` file for mac env support~~ Removed setup_java9.sh files and upgraded min Java version to 11.
- Minor fixes
- Fixed failing unit tests in `HttpMethodClientTest`

### CODE REVIEW

File: infra/docker/Dockerfile [hunk 1]
```diff
@@ -11,17 +11,33 @@ RUN apt-get update \
 RUN mkdir -p /build \
   && cd /build \
   && git clone https://github.com/NemProject/nem.git \
-  && cd nem 
+  && cd nem
```
Comment (gimre-xymcity) [inline infra/docker/Dockerfile:14]: that last cd isn't really needed

**Review by gimre-xymcity — CHANGES_REQUESTED** [review]
this change should be slightly larger:
 * we should change default compilation to java 11 (especially, that jre installed is == 11
 * and as a fallback: java8

**Comment by yilmazbahadir** [comment]
> this change should be slightly larger:
> 
> * we should change default compilation to java 11 (especially, that jre installed is == 11
> * and as a fallback: java8

sure, I'm on it now.

File: infra/docker/Dockerfile (hunk 1 above)
Comment (yilmazbahadir) [reply infra/docker/Dockerfile]: fixed

**Comment by yilmazbahadir** [comment]
As discussed in the discord channel, updated the min required Java SDK version to 11 and applied related updates in Readme files, CI scripts, pom.xml's and the dockerfile.

File: deploy/src/main/java/org/nem/deploy/NemFormatter.java [hunk 2]
```diff
@@ -14,7 +15,8 @@ public class NemFormatter extends SimpleFormatter {
 
 	@Override
 	public synchronized String format(final LogRecord record) {
-		record.setMillis(TIME_PROVIDER.getNetworkTime().getRaw() + SystemTimeProvider.getEpochTimeMillis() - TIME_ZONE_OFFSET);
```
Comment (yilmazbahadir) [inline deploy/src/main/java/org/nem/deploy/NemFormatter.java:17]: since `setMillis` method was deprecated, replaced it with `setInstant` as suggested.

**Review by gimre-xymcity — APPROVED** [review]
I was wondering if we should have fallback to java8 _somewhere_ but change itself is good so 👍

**Pull request merged**